lib/charms/layer/kubernetes_master.py

```diff
@@ -0,0 +1,65 @@
+from charmhelpers.core import hookenv
+from charms.reactive import endpoint_from_flag
+
+from charms.layer import kubernetes_common
+
+
+STANDARD_API_PORT = 6443
+
+
+def get_external_lb_endpoints():
+    """
+    Return a list of any external API load-balancer endpoints that have
+    been manually configured.
+    """
+    forced_lb_ips = hookenv.config('loadbalancer-ips').split()
+    vips = hookenv.config('ha-cluster-vip').split()
+    dns_record = hookenv.config('ha-cluster-dns')
+    if forced_lb_ips:
+        # if the user gave us IPs for the load balancer, assume
+        # they know what they are talking about and use that
+        # instead of our information.
+        return [(address, STANDARD_API_PORT) for address in forced_lb_ips]
+    elif vips:
+        return [(vip, STANDARD_API_PORT) for vip in vips]
+    elif dns_record:
+        return [(dns_record, STANDARD_API_PORT)]
```

Comment:
Previous behavior was to not use ha-cluster-vip or ha-cluster-dns unless the ha.connected flag is set. Looks like that was lost somewhere in the refactor.